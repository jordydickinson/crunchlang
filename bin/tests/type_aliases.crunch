type t = int64;

fun add(x: t, y: t): t = x + y;
