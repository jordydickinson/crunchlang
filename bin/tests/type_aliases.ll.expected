; ModuleID = 'type_aliases.crunch'
source_filename = "type_aliases.crunch"

@llvm.global_ctors = appending global [1 x void ()*] [void ()* @init.ctors]

define i64 @"add$pure"(i64 %x, i64 %y) {
entry:
  %iaddtmp = add i64 %x, %y
  ret i64 %iaddtmp
}

define void @init.ctors() {
entry:
  ret void
}
